src/Calculator.js

```diff
@@ -0,0 +1,71 @@
+const CUSTOM_EXPRESSION = Object.freeze({
+  START: "//",
+  END: "\\n",
+});
+
+export class Calculator {
+  constructor(defaultSeperators) {
+    this.seperators = defaultSeperators;
+    this.constantSeperator = ",";
+  }
+
+  extractCustomSeperator(input) {
+    if (
+      input.startsWith(CUSTOM_EXPRESSION.START) &&
+      input.includes(CUSTOM_EXPRESSION.END)
+    ) {
+      const startIndex = CUSTOM_EXPRESSION.START.length;
+      const endIndex = input.indexOf(CUSTOM_EXPRESSION.END);
+      const customSeperator = input.slice(startIndex, endIndex);
+
+      if (customSeperator.length === 0) {
+        throw new Error("[ERROR] 커스텀 구분자의 길이는 0일 수 없습니다.");
+      }
+
+      this.setCustomSeperator([
+        CUSTOM_EXPRESSION.START + customSeperator + CUSTOM_EXPRESSION.END,
+        customSeperator,
+      ]);
+    }
+  }
+
+  setCustomSeperator(customSeperators) {
+    this.seperators = [...customSeperators, ...this.seperators];
+  }
+
+  replaceAllSeperators(input) {
+    let processedInput = input;
+
+    this.seperators.forEach((seperator) => {
+      processedInput = processedInput.replaceAll(
+        seperator,
+        this.constantSeperator
+      );
+    });
```

Comment:
이 부분 로직적으로는 문제가 없지만 replaceAll이 ES2021에 추가된 기능이다보니 이전 버전과의 호환성이 떨어질 수도 있겠다는 생각이 들었어요! 이부분을 정규표현식을 사용해서 대체해보면 어떨까 제안드려보아요

```suggestion
processedInput = processedInput.split(seperator).join(this.constantSeperator);
```

Reply:
추가적으로 for loop를  사용해서 돌 때 구분자가 여러개일 경우 (이 부분을 예외처리 하셨다면 문제되진 않을 것 같아요) 문자열을 반복해서 순회해야하는 비용 문제가 생기지 않을까 하는 생각도 들었어요! 이 부분을 고려해서 적절히 수정하면 좋을 것 같아요 😄

Reply:
> 이 부분 로직적으로는 문제가 없지만 replaceAll이 ES2021에 추가된 기능이다보니 이전 버전과의 호환성이 떨어질 수도 있겠다는 생각이 들었어요! 이부분을 정규표현식을 사용해서 대체해보면 어떨까 제안드려보아요

버전 호환성은 생각지도 못했던 부분인데 좋은 지적 감사드려요! 정규표현식에 대한 학습이 아직 부족하다보니 최대한 정규표현식 없이 해결해보고 싶었는데 역시 문자열 처리에서는 정규표현식을 사용하는게 좋은 것 같네요 😂

Reply:
> 추가적으로 for loop를 사용해서 돌 때 구분자가 여러개일 경우 (이 부분을 예외처리 하셨다면 문제되진 않을 것 같아요) 문자열을 반복해서 순회해야하는 비용 문제가 생기지 않을까 하는 생각도 들었어요! 이 부분을 고려해서 적절히 수정하면 좋을 것 같아요 😄

사실 이 부분이 가장 걱정이었어요. 제 방식은 크게 두 가지 과정을 거치게 되는데

1. 모든 구분자를 하나의 기본 구분자로 치환한다. ex) `1:2,3!4` -> `1,2,3,4`
2. 기본 구분자를 기준으로 배열화 한다 ex) `1,2,3,4` -> [1, 2, 3, 4]

1번의 과정에서 구분자들을 모아둔 배열을 가지고 여러차례 순회를 돌아야하다 보니 만일 문자열이 길어지고 구분자의 종류가 많아진다면 성능문제가 반드시 발생할 수 밖에 없겠다 싶어요. 이 과정에서는 정규표현식으로 일괄 처리를 하는 것이 훨씬 좋은 방법이라고 생각이 되네요!